WireUI/Sources/WireReusableUIComponents/AccountImageView/AccountImageView.swift

```diff
@@ -0,0 +1,335 @@
+
+        // this clips a circle from the view, which gives the
+        // availability indicator view a transparent border
+        let maskLayer = CAShapeLayer()
+        maskLayer.path = maskPath.cgPath
+        maskLayer.fillRule = .evenOdd
+        accountImageView.superview?.layer.mask = maskLayer
+    }
+
+    // MARK: - Nested Types
+
+    public enum AccountType: CaseIterable {
+        /// The account image will be clipped using a circle shape.
+        case user
+        /// The account image will be clipped using a round rectangle shape.
+        case team
+    }
+}
+
+// MARK: - Convenience Init
+
+extension AccountImageView {
+
+    public convenience init(
+        accountImage: UIImage,
+        accountType: AccountType,
+        availability: Availability?
+    ) {
+        self.init()
+
+        self.accountImage = accountImage
+        self.accountType = accountType
+        self.availability = availability
+
+        updateAccountImage()
+        updateShape()
+        updateAvailabilityIndicator()
+    }
+}
+
+// MARK: - Previews
+
+@available(iOS 16.0, *)
+struct AccountImageView_Previews: PreviewProvider {
+
+    static var previews: some View {
+        Group {
+            ForEach(AccountImageView.AccountType.allCases, id: \.self) { accountType in
+
+                previewWithNavigationBar(accountType, .none)
+                    .previewDisplayName("\(accountType)")
+
+                ForEach(Availability.allCases, id: \.self) { availability in
+                    previewWithNavigationBar(accountType, availability)
+                        .previewDisplayName("\(accountType) - \(availability)")
+                }
+            }
+        }
+        .background(Color(UIColor.systemGray2))
+    }
+
+    @ViewBuilder
+    static func previewWithNavigationBar(
+        _ accountType: AccountImageView.AccountType,
+        _ availability: Availability?
+    ) -> some View {
+        let accountImage = UIImage.from(solidColor: .init(red: 0, green: 0.73, blue: 0.87, alpha: 1))
+        NavigationStack {
+            AccountImageViewRepresentable(accountImage, accountType, availability)
+                .center()
+                .scaleEffect(6)
+                .navigationTitle("Conversations")
+                .navigationBarTitleDisplayMode(.inline)
+                .toolbar {
+                    ToolbarItem(placement: .navigationBarLeading) {
+                        Button {} label: {
+                            AccountImageViewRepresentable(accountImage, accountType, availability)
+                                .padding(.horizontal)
+                        }
+                    }
+                }
+        }
+    }
+}
+
+extension View {
+
+    fileprivate func center() -> some View {
+        VStack {
+            Spacer()
+            HStack {
+                Spacer()
+                self
+                Spacer()
+            }
+            Spacer()
+        }
+    }
```

Comment:
What do you think about moving this to a new file and making it accessible internally to the module?  I'm sure that we'll collect some SwiftUI helpers and modifiers that make sense to reuse.

Reply:
I actually think this is a hack and I only used it for the preview. We should not be tempted to spread its usage, in my opinion.